desc/objectives/_free_boundary.py

```diff
@@ -686,57 +713,36 @@ def build(self, use_jit=True, verbose=1):
 
         super().build(use_jit=use_jit, verbose=verbose)
 
-    def compute(self, params, constants=None):
+    def compute(self, params=None, constants=None):
```

Comment:
Apparently you don't use params, you can get rid of it completely.
```suggestion
    def compute(self, constants=None):
```